documentation/general/dash-sonic-hld.md

```diff
@@ -342,11 +359,13 @@ DASH_ACL_RULE_TABLE:{{group_id}}:{{rule_num}}
     "priority": {{priority}}
     "action": {{action}}
     "terminating": {{bool}}
-    "protocol": {{list of protocols}}
-    "src_addr": {{list of address}}
-    "dst_addr": {{list of address}}
-    "src_port": {{list of range of ports}}
-    "dst_port": {{list of range of ports}}
+    "protocol": {{list of protocols}} (OPTIONAL)
+    "src_tag": {{list of tag name}} (OPTIONAL)
+    "dst_tag": {{list of tag name}} (OPTIONAL)
+    "src_addr": {{list of prefix}} (OPTIONAL)
```

Comment:
the documentation switched from address to prefix (prefix is definitely what we would expect). 
BUT the P4 model is still generating IP address, not prefix. could the  P4 code be fixed to match the documentation?

    /**
     * @brief Optional matched key dip
     *
     * @type sai_ip_address_t
     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
     */
    SAI_DASH_ACL_RULE_ATTR_DIP,

    /**
     * @brief Optional matched key sip
     *
     * @type sai_ip_address_t
     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
     */
    SAI_DASH_ACL_RULE_ATTR_SIP,

Reply:
Have you tried to manually define the DASH_MATCH in the file, dash_acl.p4?
```c++
#define DASH_MATCH  // Please explicitly define this line in dash-pipeline/bmv2/dash_acl.p4

#ifdef DASH_MATCH
#define LIST_MATCH list
#define RANGE_LIST_MATCH range_list
#else
#ifdef TARGET_BMV2_V1MODEL
#define LIST_MATCH optional
#define RANGE_LIST_MATCH optional
#endif // TARGET_BMV2_V1MODEL
#ifdef TARGET_DPDK_PNA
#define LIST_MATCH ternary
#define RANGE_LIST_MATCH range
#endif // TARGET_DPDK_PNA
#endif
```
If you define it, you will get the correct SAI header.
```
    /**
     * @brief List matched key dip
     *
     * @type sai_ip_prefix_list_t
     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
     */
    SAI_DASH_ACL_RULE_ATTR_DIP,

    /**
     * @brief List matched key sip
     *
     * @type sai_ip_prefix_list_t
     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
     */
    SAI_DASH_ACL_RULE_ATTR_SIP,
```

Reply:
@marian-pritsak 

is that something that could be taken care of so we align things up?
thanks

Reply:
@vincent-xs  will look at SAI Master for generation code 
Difference between APIs for SAI and bmv2 defined match types for SIP and DIP
By default not-enabled (hidden compile to check for possibly)?
@Pterosaur possibly can also check the 'define'